src/pages/track_chairs/progress-flags-page.js

```diff
@@ -33,7 +33,9 @@ class ProgressFlagsPage extends React.Component {
             flagLabel: '',
             progressFlagId: null
         }
-        this.fragmentParser = new FragmentParser();
+        
+        const fragmentParser = new FragmentParser();
+        this.progressFlagIdParam = fragmentParser.getParam('flag_id');
```

Comment:
why doing this on contructor, isnt better to have the whole behavior in a single method ? on. componentDidMount ?

Reply:
moved to componentDidMount